src/js/media-chrome-html-element.js

```diff
@@ -36,6 +37,11 @@ class MediaChromeHTMLElement extends window.HTMLElement {
       return;
     }
 
+    if (propName == 'mediaController') {
+      this[propName] = document.getElementById(newValue);
+      return;
+    }
```

Comment:
If a component is nested inside some other component's shadow DOM this won't work. Haven't a thought of a solution just yet but wanted to point it out for now.

Reply:
Yeah, that's a good point. I'd guess that'd break in the button/form relationship too.

I updated the state passing function to pierce shadow doms when open, which appears to be the most common use at least. That way as long as the host el is associated with the controller, state should reach shadow dom nodes. 

I also set `composed:true` on the custom events so they should bubble up out of a shadow dom.

That might be enough. Let me know if you still see some holes.